django/db/views.py

```diff
@@ -2112,15 +2102,13 @@ def small_molecules_for_dataset(self,dataset_id):
 #        logger.info(str(('small molecules for dataset',dataset_id,len(queryset))))
 #        return queryset
     
-    def has_plate_wells_defined(self, dataset_id):
-        res= len(DataRecord.objects.all().filter(
-            dataset_id=dataset_id).filter(plate__isnull=False).filter(well__isnull=False))>0
-        return res
+    def has_plate_wells_defined(self):
+        return DataRecord.objects.all().filter(dataset_id=self.dataset_id)\
+            .filter(plate__isnull=False).exists()
```

Comment:
this is an error - although *usually* both plate and well will be either null or set - thanks for the catch.